sbi/utils/conditional_density.py

```diff
@@ -462,15 +448,13 @@ def log_prob(self, X: Tensor, individual=False) -> Tensor:
         Returns:
             log_prob: Log probabilities at values specified by X.
         """
+        prec = self.precs@self.precs.transpose(3,2)
 
-        pdf = torch.zeros((X.shape[0], self.m.shape[0]))
-        for i in range(self.m.shape[0]):
-            pdf[:, i] = self.mulnormpdf(X, self.m[i], self.S[i]) * self.mc[0, i]
-        if individual:
-            return torch.log(pdf)
-        else:
-            log_factor = torch.log(self.leakage_correction(x=self.default_x))
-            return torch.log(torch.sum(pdf, axis=1)) - log_factor
+        self.net.eval() # leakage correction requires eval mode
+        log_factor = torch.log(self.leakage_correction(x=self.default_x))
```

Comment:
I think the leakage correction does not work for the conditioned MDNs, because it uses samples from `self` i.e. the full posterior rather then from the new, conditioned posterior.